File: credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long n = get_long("Number: ");
    int res = 0;
    int pos = 1;
    int temp = 0;
    long first_two = n;
    while (first_two >= 100)
    {
        first_two /= 10;
    }
    while (n > 0)
    {
        temp = n % 10;
        if (pos % 2 == 0)
        {
            temp *= 2;
            if (temp >= 10)
            {
                temp = temp % 10 + temp / 10;
            }
            res += temp;
        }
        else
        {
            res += temp;
        }
        n /= 10;
        pos++;
    }
    res %= 10;
    if(res == 0)
    {
        if (pos == 16 && (first_two == 34 || first_two == 37))
        {
            printf("AMEX\n");
        }
        else if ((pos == 14 || pos == 17) && (first_two / 10) == 4)
        {
            printf("VISA\n");
        }
        else if (pos == 17 && (first_two >= 51 && first_two <= 55))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: credit.py
import cs50

while True:
    number = cs50.get_string("Number: ")
    if number.isdigit():
        break
rev = number[::-1]
length = len(number)
first = int(number[0])
first2 = int(number[:2])
total = 0
for i in range(len(rev)):
    digit = int(rev[i])
    if i % 2 == 0:
        total += digit
    else:
        digit *= 2
        if digit > 9:
            digit -= 9
        total += digit
if total % 10 != 0:
    print("INVALID")
if length == 15 and (first2 == 34 or first2 == 37):
    print("AMEX")
elif length == 16 and 51 <= first2 <= 55:
    print("MASTERCARD")
elif (length == 13 or length == 16) and first == 4:
    print("VISA")
else:
    print("INVALID")
